basilisk/src/ast/interpreter/dimension-tests/test8.c

```
@@ -0,0 +1,41 @@
+      f[] = 2.;
+  }
+  foreach()
+    f[] = f[] < 3. ? 4. : f[];
+
+  /**
+  Multiplicative conditional expressions with a constant must be
+  dimensionless. So [d] should be [c] since [(b > 0 ? 1 : -2)] = [0]
+  since it is a multiplicative constant. */
+
+  double b, c = 7, d;
+  d = c*(b > 0 ? 1 : -2);
```

Cppcheck (2.10-2)

[error] Uninitialized variable

(uninitvar)